staty/codes.py

```diff
@@ -16,389 +16,405 @@
 from . import base
 from . import exceptions
 
-HTTP_STATUS_CODES = {}
-HTTP_ERROR_CODES = {}
 
+class HTTPStatusMap:
+    def __init__(self):
+        self.codes = {}
+        self.errors = {}
+        self.attribute_names = {}
 
-def register(http_status_class):
-    code = http_status_class.code
+    def __getattr__(self, item):
+        try:
+            return self.attribute_names[item]
+        except KeyError as exc:
+            raise AttributeError("{!r} object has no attribute {!r}".format(self.__class__.__name__, item)) from exc
 
-    if code in HTTP_STATUS_CODES:
-        raise exceptions.RegistrationException("Status code is already registered")
+    def register(self, http_status_class):
+        code = http_status_class.code
 
-    HTTP_STATUS_CODES[code] = http_status_class
-    if issubclass(http_status_class, base.ErrorCodeMixin):
-        HTTP_ERROR_CODES[code] = http_status_class
+        if code in self.codes:
+            raise exceptions.RegistrationException("Status code is already registered")
 
-    return http_status_class
+        self.codes[code] = http_status_class
 
+        if issubclass(http_status_class, base.ErrorCodeMixin):
+            self.errors[code] = http_status_class
 
-@register
+        status_name = "HTTP_{}_{}".format(code, "_".join(http_status_class.message.replace("-", " ").split()).upper())
```

Comment:
não queria usar o módulo re... mas se for usar já dá pra fazer quase tudo com ele...